drivers/gpio/gpio_stm32.c

```diff
@@ -91,6 +91,36 @@ static int gpio_stm32_flags_to_conf(int flags, int *pincfg)
 	return 0;
 }
 
+/**
+ * @brief Custom stm32 flags to zephyr
+ */
+static int gpio_stm32_pincfg_to_flags(int otype, int pupd, int mode,
+				      int *out_flags)
+{
+	int flags = 0;
+
+	if (mode == LL_GPIO_MODE_OUTPUT) {
```

Comment:
Is there a limitation on this API that prevents to call this API on pins not configured as a GPIO?

If not, returned mode can also be `LL_GPIO_MODE_ALTERNATE` and also `LL_GPIO_MODE_FLOATING` on F1 series.
I'm not sure about impact of `LL_GPIO_MODE_FLOATING`, but `LL_GPIO_MODE_ALTERNATE` would be quite common if this is called on a pin configureed via pinctrl API.
How is this handled in the API ?

Reply:
In zephyr there's no flag for alternate mode. My understanding is that when the alternate mode is enabled, the pin is not general-purpose, so for the GPIO subsystem, it is disabled. So this function will return the GPIO_DISCONNECTED flag.
However I have added handling the floating mode, since from the code it looks like floating should be equal to raw input mode.